### ISSUE TITLE
tweak linkerd inject output

### ISSUE THREAD

**Issue by wmorgan** [linkerd/linkerd2#1656]
While the new checks in `linked inject` are good, I think this command should not emit anything to stderr unless a problem was encountered, or a `--verbose` flag is passed in. I believe this will make it feel much more like a standard unix CLI tool.

I also think that the output should not mimic the output of `linkerd check`. E.g. rather than
```
supported: at least one resource injected..................................[warn] -- no supported objects found
```
why not just say `warning: no supported Kubernetes objects found` ?

**Comment by klingerf** [comment]
Totally agree that we should revisit this as part of updating the check output format (#1471). But one aspect of the existing output that I find especially useful is seeing the list of resources that were injected. That feels consistent with how `kubectl apply` prints the list of resources that were applied. Since the inject command must write valid yaml to stdout, then writing the list of injected resources to stderr seems like the right place to put that information.

Visually, I like seeing the list of injected resources above the list of applied resources when I pipe `linkerd inject` to `kubectl apply`:

```
Summary: 2 of 4 YAML document(s) injected
  deployment/mysql
  job/mysql-init

service "mysql" created
persistentvolumeclaim "mysql-pv-claim" created
deployment "mysql" created
job "mysql-init" created
```

We could even format the resource list to feel more like the `kubectl` output. Something like:

```
service "mysql" skipped
persistentvolumeclaim "mysql-pv-claim" skipped
deployment "mysql" injected
job "mysql-init" injected

service "mysql" created
persistentvolumeclaim "mysql-pv-claim" created
deployment "mysql" created
job "mysql-init" created
```

I agree that the `linkerd check` style output that precedes the resource list is a bit busy, especially when everything is working as expected:

```
hostNetwork: pods do not use host networking...............................[ok]
sidecar: pods do not have a proxy or initContainer already injected........[ok]
supported: at least one resource injected..................................[ok]
udp: pod specs do not include UDP ports....................................[ok]
```

In which case I'd be ok with switching that part of the output to only print errors.

**Comment by siggy** [comment]
Fixed via #2087.
